src/libraries/System.ServiceModel.Syndication/src/System/ServiceModel/XmlBuffer.cs

```diff
@@ -86,7 +86,21 @@ public void Close()
             _bufferState = BufferState.Reading;
             _buffer = new byte[_stream.Length];
             _stream.Position = 0;
-            _stream.Read(_buffer, 0, _buffer.Length);
+
+#if NET7_0_OR_GREATER
+            _stream.ReadExactly(_buffer);
+#else
+            int totalRead = 0;
+            while (totalRead < _stream.Length)
+            {
+                int bytesRead = _stream.Read(_buffer, totalRead, _stream.Length - totalRead);
```

Comment:
FYI build failure: `error CS1503: (NETCORE_ENGINEERING_TELEMETRY=Build) Argument 3: cannot convert from 'long' to 'int'`